bonsai/rest-api-mock/src/prover.rs

```diff
@@ -102,7 +102,9 @@ impl Prover {
                         anyhow::anyhow!("failed to build executor environment: {:?}", e)
                     })?;
                 let mut exec = Executor::new(env, mem_img)?;
-                let session = exec.run()?;
+                let session = exec.run().map_err(|e| {
+                    anyhow::anyhow!("Executor failed to generate a successful session: {:?}", e)
+                })?;
```

Comment:
Anyhow provides the [context function](https://docs.rs/anyhow/latest/anyhow/trait.Context.html) to add additional information like this to an error. It also preserves more metadata.
```suggestion
                let session = exec.run().context("Executor failed to generate a successful session)?;
```